channeldb/db.go

```diff
@@ -1367,6 +1405,61 @@ func (d *DB) syncVersions(versions []version) error {
 	}, func() {})
 }
 
+// applyOptionalVersions takes a config to determine whether the optional
+// migrations will be applied.
+//
+// NOTE: only support the prune_revocation_log optional migration atm.
+func (d *DB) applyOptionalVersions(cfg OptionalMiragtionConfig) error {
```

Comment:
So we'd thread through that quit channel _here_ to make the migration resumable.

Reply:
So I guess to achieve this, we'll pass the `ctx` used in `BuildDatabase` through `BuildDatabase -> CreateWithBackend -> applyOptionalVersions -> version.migration` and catch the cancel signal there?